### PR TITLE
`Popover`: update `positionToPlacement` types

### PR DESCRIPTION
**Pull request by brookewp**
## What?

This change is based on a[ diff ](https://github.com/WordPress/gutenberg/pull/48440#discussion_r1312093852)suggested by @mirka after reviewing upcoming changes to Tooltip in #48440. 

Last year, a utility function `positionToPlacement` was created to assist in converting values for the legacy `position` prop to the newer `placement` prop. #44377

The `position` prop supported `overlay`, but `placement` does not. However, @ciampo shared that `positionToPlacement` doesn't return `overlay` and proposed [a similar diff](https://github.com/WordPress/gutenberg/pull/48440#discussion_r1147927828) as well. 

## Why?

The `overlay` type from `position` isn't part of `placement` from `floating-ui`, so we see TS errors in the aforementioned Tooltip. With more upcoming migrations, this may cause further TS errors in other components. Since `overlay` isn't returned, we can remove it. 

## How?

Replace `PopoverProps['placement']` type with `Placement` from `floating-ui` to remove type `overlay`. 

## Testing Instructions

CI checks pass

### CODE REVIEW

**Comment by brookewp** [comment]
@mirka - does it make sense to import `Placement` into the test to replace it there as well? 
https://github.com/WordPress/gutenberg/blob/e2941e9741bb8e21f0d7965d6bf70d43113bade2/packages/components/src/popover/test/index.tsx#L26

**Review by chad1008 — COMMENTED** [review]
changes look good to me, I'd say we're just waiting on the failing/flaky CI?

**Review by ciampo — APPROVED** [review]
LGTM 🚀 

My initial reasoning for using `PopoverProps[ 'placement' ]` was to avoid referencing `floating-ui` directly, but I guess it doesn't make much of a difference, and it actually caused issues once the new `overlay` custom placement was added to `Popover`.

Could you add a CHANGELOG entry too?

Good to merge once CI passes (rebasing/mergin trunk may help)

Thank you for working on this!

**Pull request merged**
